python/ray/_private/accelerators/nvidia_gpu.py

```diff
@@ -0,0 +1,153 @@
+import re
+import os
+import sys
+import logging
+import subprocess
+import importlib
+from typing import Optional, List
+
+try:
+    import GPUtil
+except ImportError:
+    pass
+
+from ray._private.accelerators.accelerator import Accelerator
+
+logger = logging.getLogger(__name__)
+
+CUDA_VISIBLE_DEVICES_ENV_VAR = "CUDA_VISIBLE_DEVICES"
+NOSET_CUDA_VISIBLE_DEVICES_ENV_VAR = "RAY_EXPERIMENTAL_NOSET_CUDA_VISIBLE_DEVICES"
+
+# TODO(Alex): This pattern may not work for non NVIDIA Tesla GPUs (which have
+# the form "Tesla V100-SXM2-16GB" or "Tesla K80").
+NVIDIA_GPU_NAME_PATTERN = re.compile(r"\w+\s+([A-Z0-9]+)")
+
+
+class NvidiaGPUAccelerator(Accelerator):
+    """Nvidia GPU accelerators."""
+
+    @staticmethod
+    def get_resource_name() -> str:
+        return "GPU"
+
+    @staticmethod
+    def is_available() -> bool:
+        return NvidiaGPUAccelerator.get_num_accelerators() > 0
+
+    @staticmethod
+    def get_visible_accelerator_ids_env_var() -> str:
+        return CUDA_VISIBLE_DEVICES_ENV_VAR
+
+    @staticmethod
+    def get_visible_accelerator_ids() -> Optional[List[str]]:
+        cuda_visible_devices = os.environ.get(CUDA_VISIBLE_DEVICES_ENV_VAR, None)
+
+        if cuda_visible_devices is None:
+            return None
+
+        if cuda_visible_devices == "":
+            return []
+
+        if cuda_visible_devices == "NoDevFiles":
+            return []
+
+        return list(cuda_visible_devices.split(","))
```

Comment:
Do we need regex matching here?

Reply:
No? I just copy paste the existing code.